src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -1118,6 +1120,26 @@ public JsValue PutDocument(JsValue self, JsValue[] args)
                 }
             }
 
+            private string GenerateIdForShard(JsValue[] args, string id)
+            {
+                if (id?[^1] != _database.IdentityPartsSeparator)
+                    return id;
+
+                var originalId = (args[1].AsObject() as BlittableObjectInstance)?.DocumentId;
+                var builder = new StringBuilder(id);
+                var index = originalId.IndexOf('$');
+                if (index != -1)
+                    originalId = originalId[index..originalId.Length];
+                else
+                    builder.Append('$');
```

Comment:
You are writing `$` and then adding it again below

Reply:
Right, just like `StickyId` with random string - it's been removed in `PutDocument->BuildDocumentId`